### PR TITLE
just extend dataset and query fix

### PR DESCRIPTION
**Pull request by masterhood13**
# Pull Request Title

## Description

Please include a summary of the changes and the related issue number. Provide relevant motivation and context. If this fixes a bug or resolves a feature, include a link to the related issue.

- **Issue Number**: [Issue #]
- **Description of changes**: Explain the code, algorithms, or models modified/added.

## Type of Change

Please delete options that are not relevant:

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Code refactor
- [x] Performance improvement

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Include any relevant details for configuration.

- **Unit Tests Added**: [Yes/No]
  - If yes, describe the tests and if they were successful.
- **Model Accuracy Improvement**: [Yes/No]
  - If yes, provide before and after metrics of the prediction model accuracy (e.g., accuracy, precision, recall, etc.)

**Test Configuration**:
- Python version: [e.g., 3.9.19]
- ML Framework/Library: [e.g., XGBoost]
- Dataset: [e.g., all_data.csv]

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

## Additional Context or Screenshots (if applicable)

Include any additional context or relevant screenshots related to the prediction models or new features. If the change impacts the Dota 2 match prediction outcomes, provide graphs or results showing the impact.

### CODE REVIEW

File: structure/struct.py [hunk 1]
```diff
@@ -604,8 +604,12 @@ def get_league_matches(self):
                 match = Match(
                     match_id, radiant_team_id, dire_team_id, self.league_id, radiant_win
                 )
-                match.get_match_data()
-                self.add_match(match)
+                try:
+                    match.get_match_data()
+                    self.add_match(match)
+                except (TypeError, KeyError):
+                    # done ofr skip broken data
```
Comment (vkatsuba) [inline structure/struct.py:611]: ofr?

Comment (masterhood13) [reply structure/struct.py:611]: need to build new version, becuse NN retraind on extended dataset. need to rebuild it after merge

Comment (masterhood13) [reply structure/struct.py:611]: i mean on docker

Comment (masterhood13) [reply structure/struct.py]: fixed

File: structure/struct.py [hunk 2]
```diff
@@ -604,8 +604,12 @@ def get_league_matches(self):
                 match = Match(
                     match_id, radiant_team_id, dire_team_id, self.league_id, radiant_win
                 )
-                match.get_match_data()
-                self.add_match(match)
+                try:
+                    match.get_match_data()
+                    self.add_match(match)
+                except (TypeError, KeyError):
```
Comment (vkatsuba) [inline structure/struct.py:610]: It is possible to cover it somehow by some tests?

Comment (masterhood13) [reply structure/struct.py:610]: it is a floating thing, somitimse opendota return broken data, which not contains all needed info, it occures once pre 500 matches, so i added it to not braek the runtime

**Review by vkatsuba — APPROVED** [review]

**Pull request merged**